api/src/modules/projects/projects.service.ts

```diff
@@ -185,6 +188,17 @@ export class ProjectsService extends AppBaseService<
       : project.countryId
       ? await this.countriesService.getById(project.countryId)
       : undefined;
-    return planningArea;
+    return planningArea
+  }
+
+  async actionAfterCreate(
+    model: Project,
+    createModel: CreateProjectDTO,
+    info?: AppInfoDTO,
+  ): Promise<void> {
+    if (createModel.planningUnitAreakm2 && createModel.planningUnitGridShape) {
```

Comment:
Ideally, it should be transparent, the reality is that right now it will only take care of regular generated grids (as we currently do not support user grid uploads). in my mind these are the possibilities:
An area(extent)/region/custom grid is not supplied -> nothing is triggered 
A custom extent (area) is supplied, but not planning unit area type + grid size or user grid -> nothing is triggered
A planning unit area type + grid size is supplied but  not an area(extent)/region/custom grid is provided -> nothing is triggered

An area(extent)/region is provided + a planning unit area type(regular grid hexagon or square or irregular) + grid size -> PU grid generation is triggered.

A custom grid uploaded by the user is provided -> Regular grid creation should not be triggered, but we should upload that grid as a custom one set the PlanningUnitGridShape.fromShapefile and infer the extent and /or admin regions from there. Also, have a way to retrieve that user data as part of the project. 

* the area extent can be also uploaded by the user but is only a geometry that represents a custom extent.

And yes the same logic should go in after the update, + a check to invalidate everything if this is triggered. 
And also we must do all this checks/validations before we trigger this.

But as you mention this looks like work for a next pr :D (i will put this exact thing in jira so we dont forget about all things that need to happen)